kubert/src/runtime.rs

```diff
@@ -126,6 +207,12 @@ impl<S> Builder<S> {
         self
     }
 
+    /// Configures the runtime to record watch metrics with the given registry
+    pub fn with_metrics(mut self, metrics: ResourceWatchMetrics) -> Self {
```

Comment:
As I mentioned [here](https://github.com/olix0r/kubert/pull/243#discussion_r1566506798), I don't think that this type should be part of the public library interface: we want a single public RuntimeMetrics type that is responsible for instrumenting anything about the runtime we care to track, even if for now it only holds a ResourceWatchMetrics internally.

This way:
* we can add more metrics without changing any part of the public api
* we can change the internal registry structure without impacting the public API (and therefore, the ResourceWatchMetrics type should not be public).